ivadomed/loader/generalized_loader_configuration.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import annotations
```

Comment:
General comment about this file.

From my understanding, it will be used in future PR to simplified the dataset constructor parameters.
* I find the name "loader" to be misleading because we also gave a concept of "loader parameters" in the config files which is only a part of all of these parameters here. I think `generalized_loading_configuration` would alleviate this ambiguity.
* Where are the reference for the default value? Is it in `config_default.json` or in the current definitions of `MRI2D` and `MRI3D` datasets?
* How de we plan to keep this up-to-date during the transition? I would be cautious to introduce a file that supercedes other parameters without a control mecanism.

I think I don't understand correctly the purpose of this file in the "big picture" so I don't know exactly how to review it.

Reply:
Valid concern. Renaming adopted. 

As for the default values, they are taken from all existing constructor function defaults which assigned these values already as they: 

You can see it here: 

![image](https://user-images.githubusercontent.com/6720068/211382520-9aa46f5b-2ddf-4648-b421-7944eeea9424.png)

It should have no impact on all functions because... the default we use is the same as the mri2d and mri3d datasets. If there is upstream overwrites, it gets past over and has precedent. if not those defaults have been used for a while. 

It is superseding in the sense that it overwrites the default for MRI2D/MRI3D, which we may eventually default to mandatorily require parameters in the future to avoid duplication of defaults but not needed to determined that at the moment. 

As for all existing changes, it does not affect anything outside of V2 loader pathway as this is not used anywhere else.

Reply:
Thanks for the detailed explanation.
My concern is mainly for `length`, `stride` and `slice_axis` which have different default values in 2D and 3D in the screenshot above.
In 3D, length is always used because we always use subvolumes (or patches).
In 2D, we do both "patch" and "slice" (no-patch) training/testing. And the "patch" way is triggered by the presence of the length param.
When I tested the PR #1201, I had no issues with this param so I think it's OK but this is something to keep in mind.

Reply:
So I just tested again in PR #1201 and the default values of length here (in 3D) don't seem to impact the behavior in 2D with or without patches.

However, as described in this [comment](https://github.com/ivadomed/ivadomed/pull/1201/files#r1069617005) in PR #1201, the prototype does not work in 3D at the moment.

Reply:
To move forward, a note in the class should be added for 3D scenarios to revisit this and is not lost in the complete integration process.